src/js/tools/pen.js

```diff
@@ -66,25 +66,31 @@ define(function (require, exports, module) {
                     return descriptor.playObject(optionsObj);
                 });
 
-        if (this.vectorMode) {
-            var selectVectorMask = descriptor.playObject(vectorMaskLib.activateVectorMaskEditing());
-            return Promise.join(setPromise, selectVectorMask);
+        var deleteFn = function (event) {
+            event.stopPropagation();
+
+            return PS.performMenuCommand(_CLEAR_PATH)
```

Comment:
1. You don't need to return this promise, because the shortcut handler ignores the return value.
2. I think it would be preferable to execute (i.e., enqueue) the `menu.native` flux action here.